cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java

```diff
@@ -0,0 +1,115 @@
+package com.cloudant.sync.replication;
+
+import com.cloudant.sync.notifications.ReplicationCompleted;
+import com.cloudant.sync.notifications.ReplicationErrored;
+import com.google.common.eventbus.Subscribe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class ReplicationPolicyManager {
+
+    private final List<Replicator> replicators = new ArrayList<Replicator>();
+    private ReplicationListener replicationListener;
+    private ReplicationsCompletedListener mReplicationsCompletedListener;
+
+    public interface ReplicationsCompletedListener {
+        void allReplicationsCompleted();
+
+        void replicationCompleted(int id);
+
+        void replicationErrored(int id);
+    }
+
+    private class ReplicationListener {
+
+        Set<Replicator> replicatorsInProgress;
+
+        ReplicationListener() {
+            replicatorsInProgress = Collections.synchronizedSet(new HashSet<Replicator>());
+        }
+
+        void add(Replicator replicator) {
+            replicatorsInProgress.add(replicator);
```

Comment:
Possibly use just `synchronized(replicatorsInProgress)` and avoid `synchronizedSet`?  Though the documentation for the `synchronizedSet` call suggests you can use `synchronized` on a `synchronizedSet` returned set.